system/utils.py

```diff
@@ -533,13 +553,101 @@ def check_ledger_sync():
     demote_data = json.dumps({'alias': alias, 'services': []})
     demote_req = await ledger.build_node_request(trustee_did, target_did, demote_data)
     demote_res = json.loads(await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, demote_req))
+    while demote_res['op'] != 'REPLY':
```

Comment:
What if request gets rejected for some reason? Will we just hang here forever or there is some timeout somewhere?

Reply:
Removed this from demote/promote helpers and wrapped them into 'eventually'.